zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/PartitionConfigurationManager.java

```diff
@@ -65,4 +76,118 @@ ActorFuture<Void> disableExporter(final String exporterId) {
 
     return exportedDisabled;
   }
+
+  /**
+   * Enables the given exporter in this partition. This information will be added to the
+   * PartitionContext. The exporter will be initialized from the exporter with the id specified by
+   * initializeFrom if it is not null.
+   *
+   * <p>This method access and updates PartitionContext. Hence, it must be executed from the
+   * ZeebePartition actor.
+   *
+   * @param exporterId id of the exporter to enable
+   * @param metadataVersion the version of the metadata to set in the exporter state
+   * @param initializeFrom the id of the exporter to initialize from. Can be null.
+   * @return the future that completes when the exporter is enabled
+   */
+  ActorFuture<Void> enableExporter(
+      final String exporterId, final long metadataVersion, final String initializeFrom) {
+    final var exporterEnabled = executor.<Void>createFuture();
+
+    final var exporterDescriptor = validateAndGetDescriptor(exporterId, initializeFrom);
+    exporterDescriptor.ifRightOrLeft(
+        descriptor ->
+            enableExporter(
+                exporterId, metadataVersion, initializeFrom, descriptor, exporterEnabled),
+        throwable -> {
+          logger.warn("Failed to enable exporter {}", initializeFrom, throwable);
+          exporterEnabled.completeExceptionally(throwable);
+        });
+
+    return exporterEnabled;
+  }
+
+  private Either<Exception, ExporterDescriptor> validateAndGetDescriptor(
+      final String exporterId, final String initializeFrom) {
+    if (initializeFrom == null) {
+      return getExporterDescriptor(exporterId);
+    }
+
+    return getExporterDescriptor(exporterId)
+        .flatMap(
+            descriptor ->
+                getExporterDescriptor(initializeFrom)
+                    .flatMap(
+                        otherDescriptor -> verifyValidInitialization(descriptor, otherDescriptor))
+                    .map(ignore -> descriptor));
+  }
+
+  private void enableExporter(
+      final String exporterId,
+      final long metadataVersion,
+      final String initializeFrom,
+      final ExporterDescriptor descriptor,
+      final ActorFuture<Void> exporterEnabled) {
+    // Do not re-enable the exporter if the operation is retried
+    final ExporterState exporterState =
+        context.getDynamicPartitionConfig().exporting().exporters().get(exporterId);
+    if (exporterState != null && exporterState.state() == State.ENABLED) {
+      logger.trace("Exporter {} is already enabled", exporterId);
+      exporterEnabled.complete(null);
+      return;
+    }
+
+    final var updatedConfig =
+        context
+            .getDynamicPartitionConfig()
+            .updateExporting(
+                config -> config.enableExporter(exporterId, initializeFrom, metadataVersion));
+    context.setDynamicPartitionConfig(updatedConfig);
```

Comment:
:question: Shouldn't we set this only if the enabling worked? Because if we set it now, and it fails, we won't retry as per the short cut above?

Reply:
We have to set it before to handle in any concurrent role transitions. If we do not set it, it can happen that we enable the exporter, then a role change happened, we install ExporterDirector with the exporter disabled, and then only set the config. 

I think then this answer's above question. We should remove this shortcut and make it a `noop` in ExporterDirector when retried.